src/Nullable.ts

```diff
@@ -0,0 +1,108 @@
+ * operator, and the nullish-safe accessor `?.` operator. However, those operators lend
+ * themselves primarily to OO-style programming, not function pipelining or coposition.
+ * These functions are curried and are designed to be used with right-to-left function
+ * composition like `pipe` and `flow`.
+ *
+ * @example
+ * declare const str: Nullable<string>
+ *
+ * pipe(
+ *     str,
+ *     Nullable.map(s => `¡${s}!`),
+ *     Nullable.map(String.reverse),
+ *     Nullable.defaultWith(() => "")
+ * ) // => "!yoha¡", if str is "ahoy"; "" if str is null or undefined
+ */
+export type Nullable<A extends {}> = A | null | undefined
+
+/**
+ * Get an `EqualityComparer` that considers `null` and `undefined` equivalent, and
+ * compares non-nullish values based on the given `EqualityComparer`.
+ *
+ * @category Utils
+ *
+ * @example
+ * const { equals } = Nullable.getEqualityComparer(EqualityComparer.Number)
+ * equals(null, undefined) // => true
+ * equals(3, undefined) // => false
+ * equals(null, 4) // => false
+ * equals(4, 4) // => true
+ * equals(4, 5) // => false
+ */
+const getEqualityComparer = <A extends {}>({ equals }: EqualityComparer<A>) =>
+    EqualityComparer.ofEquals<Nullable<A>>(
+        (a1, a2) =>
+            (a1 == null && a2 == null) || (a1 != null && a2 != null && equals(a1, a2))
+    )
+
+/**
+ * Similar to `Option.defaultValue`. If the given nullable value is nullish, returns
+ * the given fallback/default value. If the given nullable value is non-nullish,
+ * returns the value.
+ *
+ * @param a The default/fallback value to use.
+ *
+ * @returns A non-nullable value.
+ *
+ * @category Utils
+ */
+const defaultValue =
+    <A extends {}>(a: A) =>
+    (nullable: Nullable<A>): NonNullable<A> =>
+        nullable != null ? nullable : a
+
+/**
+ * Similar to `Option.defaultWith`. If the given nullable value is nullish,
+ * computes the fallback/default value using the given function. If the given
+ * nullable value is non-nullish, returns the value.
+ *
+ * @param f The function to use to compute the default/fallback value.
+ *
+ * @returns A non-nullable value.
+ *
+ * @category Utils
+ */
+const defaultWith =
+    <A extends {}>(f: () => A) =>
+    (nullable: Nullable<A>): NonNullable<A> =>
+        nullable != null ? nullable : f()
+
+/**
+ * Similar to `Option.map`. Uses the given function to map the nullable value
+ * if it is non-nullish. Passes through nullish values unchanged.
+ *
+ * @category Mapping
+ */
+const map =
+    <A extends {}, B extends {}>(f: (a: A) => B) =>
+    (nullable: Nullable<A>): Nullable<B> =>
+        nullable != null ? f(nullable) : nullable
+
+/**
+ * Similar to `Option.bind`. Maps the nullable value using a function that itself
+ * returns a possibly nullish value, and flattens the result.
+ *
+ * @category Mapping
+ */
+const bind =
+    <A extends {}, B extends {}>(f: (a: A) => Nullable<B>) =>
+    (nullable: Nullable<A>): Nullable<B> =>
+        nullable != null ? f(nullable) : nullable
```

Comment:
:question: This and the `map` implementation can be the same because there's no actual flattening to do, right? Since `Option<A>` can be `A` and doesn't have some property that stores the underlying `A` (like `some`), right?

Reply:
I actually tried this but I got this bunch of gobbledy-gook back :shrug: I figure it's just one line of code in the actual bundle 

```
Type '<A extends {}, B extends {}>(f: ((a: A) => B) | ((a: A) => Nullable<B>)) => (nullable: Nullable<A>) => Nullable<B>' is not assignable to type '<A extends {}, B extends {}>(f: (a: A) => Nullable<B>) => (nullable: Nullable<A>) => Nullable<B>'.
  Call signature return types '(nullable: Nullable<A>) => Nullable<{}>' and '(nullable: Nullable<A>) => Nullable<B>' are incompatible.
    Type 'Nullable<{}>' is not assignable to type 'Nullable<B>'.
      Type '{}' is not assignable to type 'Nullable<B>'.ts(2322)
```

Reply:
https://github.com/taylorn16/fp-toolkit/pull/21/commits/486cdbf07f3e4d23c6335dbf976713dbd5baf8a7

Reply:
Oh I wasn't suggesting a change, just trying to make sure I understood what was going on with the types. 😅

Reply:
>because there's no actual flattening to do, right?

In that case, yes this is exactly correct.